internal/service/certificatesigningrequest.go

```diff
@@ -268,42 +258,48 @@ func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, r
 func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, request server.ApproveCertificateSigningRequestRequestObject) (server.ApproveCertificateSigningRequestResponseObject, error) {
 	orgId := store.NullOrgId
 
-	csr, err := h.store.CertificateSigningRequest().Get(ctx, orgId, request.Name)
+	storeCsr := h.store.CertificateSigningRequest()
+	csr, err := storeCsr.Get(ctx, orgId, request.Name)
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		} else {
-			return nil, err
 		}
+		return nil, err
 	}
 
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
+	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestApproved) {
+		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been approved"}, nil
+	}
+
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
+	if err != nil {
+		return nil, err
+	}
 
 	approvedCondition := api.Condition{
 		Type:    api.CertificateSigningRequestApproved,
 		Status:  api.ConditionStatusTrue,
 		Reason:  "Approved",
 		Message: "Approved",
 	}
-	err = h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, []api.Condition{approvedCondition})
-	if err != nil {
-		if errors.Is(err, flterrors.ErrResourceNotFound) {
-			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		}
-		return nil, err
-	}
 
-	err = signApprovedCertificateSigningRequest(h.ca, csr)
-	if err != nil {
-		if errors.Is(err, flterrors.ErrResourceNotFound) {
-			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		}
+	csr.Status.Certificate = &signedCert
+	csr.Status.Conditions = append(csr.Status.Conditions, approvedCondition)
```

Comment:
```suggestion
	api.SetStatusCondition(&csr.Status.Conditions, approvedCondition)
```

Reply:
It didn't like this last time, but some other code has changed so I'll try again

Reply:
@avishayt fixed